### PR TITLE
Major update of documentation

### PR DESCRIPTION
**Pull request by kampersanda**
This PR greatly enhances the front page of the documentation.

### CODE REVIEW

**Comment by kampersanda** [comment]
@hirosassa Could you review the new document when you have time?

It can be compiled and opened with the following command:

```
RUSTDOCFLAGS="--html-in-header katex.html" cargo doc --no-deps --open
```

**Comment by hirosassa** [comment]
OK! I'll check this.

File: src/wavelet_matrix.rs [hunk 1]
```diff
@@ -378,6 +378,13 @@ impl WaveletMatrix {
     /// - `ranges`: Ranges to be searched.
     /// - `k`: Occurrence threshold.
     ///
+    /// # Complexity
+    ///
+    /// - $`O( \min(\sigma, j_1 - i_1, \dots, j_r - i_r ) )`$ for `ranges` is $`[(i_1,j_1),\dots,(i_r,j_r)]`$.[^intersect]
+    ///
+    /// [^intersect]: A tighter bound is analyzed in the paper: Gagie, Travis, Gonzalo Navarro, and Simon J. Puglisi.
+    /// "New algorithms on wavelet trees and applications to information retrieval." Theoretical Computer Science 426 (2012): 25-41.
+    ///
```
Comment (hirosassa) [inline src/wavelet_matrix.rs:387]: :+1:

File: src/lib.rs [hunk 2]
```diff
@@ -1,28 +1,200 @@
-//! # `sucds`: Succinct data structures in Rust
+//! # Succinct data structures in Rust
 //!
-//! `sucds` contains some [succinct data structures](https://en.wikipedia.org/wiki/Succinct_data_structure) written in Rust.
+//! Sucds is a collection of [succinct data structures](https://en.wikipedia.org/wiki/Succinct_data_structure),
+//! powerful tools to store a variety of data structures in compressed space and
+//! quickly perform operations on the compressed data.
+//!
+//! ## Design policy
```
Comment (hirosassa) [inline src/lib.rs:7]: I strongly agree with this policy.

**Review by hirosassa — APPROVED** [review]
Looks great!

**Comment by kampersanda** [comment]
@hirosassa Thanks!

**Pull request merged**